miri-script/src/commands.rs

```diff
@@ -0,0 +1,652 @@
+                "cargo {toolchain} run {extra_flags...} --manifest-path {miri_manifest} -- {miri_flags} {flags...}"
+            ).run()?;
+        }
+        Ok(())
+    }
+
+    fn fmt(&self, flags: &[OsString]) -> Result<()> {
+        let toolchain = self.get_toolchain_override();
+        let config_path = path!(self.miri_dir / "rustfmt.toml");
+        let sh = self.shell_with_miri_env()?;
+        for item in WalkDir::new(&self.miri_dir).into_iter().filter_entry(|entry| {
+            let name: String = entry.file_name().to_string_lossy().into();
+            let ty = entry.file_type();
+            if ty.is_file() {
+                name.ends_with(".rs")
+            } else {
+                // dir or symlink
+                &name != "target"
+            }
+        }) {
+            let item = item.unwrap(); // Should never panic as we've already filtered out failed entries.
+            if item.file_type().is_file() {
+                let path = item.path();
+                cmd!(sh, "rustfmt {toolchain} --edition=2021 --config-path {config_path} {flags...} {path}").quiet().run()?;
+            }
+        }
+        Ok(())
+    }
+
+    fn clippy(&self, flags: &[OsString]) -> Result<()> {
+        let toolchain_modifier: OsString = self.get_toolchain_override();
+        let extra_flags = &self.cargo_extra_flags;
+        let miri_manifest = path!(self.miri_dir / "Cargo.toml");
+        let sh = self.shell_with_miri_env()?;
+        cmd!(sh, "cargo {toolchain_modifier} clippy {extra_flags...} --manifest-path {miri_manifest} --all-targets -- {flags...}").run()?;
+        Ok(())
+    }
+
+    fn cargo(&self, flags: &[OsString]) -> Result<()> {
+        // We carefully kept the working dir intact, so this will run cargo *on the workspace in the
+        // current working dir*, not on the main Miri workspace. That is exactly what RA needs.
+        let toolchain_modifier = self.get_toolchain_override();
+        let sh = self.shell_with_miri_env()?;
+        cmd!(sh, "cargo {toolchain_modifier} {flags...}").run()?;
+        Ok(())
+    }
+    fn many_seeds(&self, command: &[OsString], seed_start: u64, seed_count: u64) -> Result<()> {
+        let seed_end = seed_start + seed_count;
+        assert!(!command.is_empty());
+        let (command_name, trailing_args) = command.split_first().unwrap();
+        let sh = shell_with_parent_env()?;
+        for seed in seed_start..seed_end {
+            println!("Trying seed: {seed}");
+            let mut miriflags = std::env::var_os("MIRIFLAGS").unwrap_or_default();
+            miriflags.push(format!(" -Zlayout-seed={seed} -Zmiri-seed={seed}"));
+            let status =
+                cmd!(sh, "{command_name} {trailing_args...}").env("MIRIFLAGS", miriflags).run();
+            if status.is_err() {
+                println!("Failing seed: {seed}");
+                break;
+            }
+        }
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use shell_words::split;
```

Comment:
Why are you testing another crate's function here...?

Reply:
I've left that in while looking for a way to split the shell words. I'll remove that, I am sorry. :(